std/assembly/dataview.ts

```diff
@@ -16,9 +16,11 @@ export class DataView {
   constructor(
     readonly buffer: ArrayBuffer,
     readonly byteOffset: i32 = 0,
-    readonly byteLength: i32 = i32.MAX_VALUE,
+    readonly byteLength: i32 = buffer.byteLength - byteOffset,
```

Comment:
```diff
diff --git a/std/assembly/dataview.ts b/std/assembly/dataview.ts
index f1a8ebe..ef5df88 100644
--- a/std/assembly/dataview.ts
+++ b/std/assembly/dataview.ts
@@ -16,8 +16,10 @@ export class DataView {
   constructor(
     readonly buffer: ArrayBuffer,
     readonly byteOffset: i32 = 0,
-    readonly byteLength: i32 = buffer.byteLength - byteOffset,
+    readonly byteLength: i32 = i32.MIN_VALUE,
   ) {
+    if (byteLength === i32.MIN_VALUE) byteLength = buffer.byteLength - byteOffset
+
     if (byteOffset < 0) throw new RangeError("byteOffset cannot be negative");
     if (byteLength < 0) throw new RangeError("byteLength cannot be negative");
     if (byteOffset + byteLength > buffer.byteLength) throw new RangeError("Length out of range of buffer");
```

Is this an acceptable compromise?

It won't give a proper error if someone passes in `i32.MIN_VALUE` as the `byteLength`, but I think we have to choose one i32 value to mean "nothing was passed"

(btw. creating a `DataView` with a `byteLength` of `0` is totally valid)

Reply:
What about this?
```ts
byteLength = min(buffer.byteLength - byteOffset, byteLength);
```

Reply:
I was thinking that it would be preferable to throw if a too long byteLength was passed, rather than silently truncating the length of the `DataView`. That is how the spec says as well:

https://www.ecma-international.org/ecma-262/6.0/#sec-dataview-buffer-byteoffset-bytelength